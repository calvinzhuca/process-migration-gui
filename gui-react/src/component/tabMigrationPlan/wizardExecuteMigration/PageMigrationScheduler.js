import React, { Component } from 'react';
import Datetime from 'react-datetime';
import moment from 'moment';

export default class PageMigrationScheduler extends Component {
    constructor(props){
        super(props);
        //this.disableScheduleTime();
        this.state = {
          dateTimeInput:true
        };

    }

    setCallbackUrl = (event) =>{
        this.props.setCallbackUrl(event.target.value);
    }

    disableScheduleTime = () =>{
        console.log('disable scheduleTime');
//        var scheduleTime = document.getElementById("PageMigrationScheduler_scheduleTime")
//        console.log('disable scheduleTime' + scheduleTime.value);
//        scheduleTime.disabled = true;
//        scheduleTime.value='';

        this.setState({
            dateTimeInput:false
         });
         this.props.setScheduleStartTime('');
    }

    enableScheduleTime = () =>{
        console.log('enable scheduleTime');
        //document.getElementById("PageMigrationScheduler_scheduleTime").disabled = false;
        this.setState({
            dateTimeInput:true
         });
    }

    handleDateTimeInput = (inputMoment) =>{
        if (moment(inputMoment,"YYYY-MM-DDTHH:mm:ss", true).isValid()){
            console.log('handleDateTimeInput, valid moment: ' + inputMoment.toDate());
    //        this.props.setScheduleStartTime(inputMoment.format("YYYY/MM/DD hh:mm:ss a"));
    //2019-01-30T13:00:00-05:00        this.props.setScheduleStartTime(inputMoment.format(moment.ISO_DATE_TIME));
    //2019-01-30T18:00:00.000Z         this.props.setScheduleStartTime(inputMoment.toISOString());
    //2019-01-30T18:00:00.000Z            this.props.setScheduleStartTime(inputMoment.toDate());
            this.props.setScheduleStartTime(inputMoment.toDate());
        }else{
            //just ignore the user input
            console.log('handleDateTimeInput, not valid moment: ' + inputMoment);
        }
    }

    validDate = (current) =>{
        var yesterday = Datetime.moment().subtract( 1, 'day' );
        return current.isAfter(yesterday);
    }

  render() {

    return (
        <div className="form-horizontal">

            <table border="0" width="100%" height="30%">
              <tbody>
              <tr>
                  <td width="30%"/>
                  <td width="40%" align="left">
                      <label data-testid="testid_callback">Callback URL: <input type="text" name="callbackUrl" onChange={this.setCallbackUrl} value={this.props.callbackUrl}/></label>
                </td>
                <td width="40%"/>
              </tr>
                <tr>
                    <td width="30%"/>
                    <td width="40%" align="left">
                        <label data-testid="testid_syncMode"><input type="radio" name="timeType" value="1" onClick={this.disableScheduleTime}/>Run Migration Now</label>
                    </td>
                    <td width="30%"/>
                </tr>
                <tr>
                    <td width="30%"/>
                    <td width="40%" align="left">
                        <label data-testid="testid_asyncMode"><input type="radio" name="timeType" value="2" defaultChecked onClick={this.enableScheduleTime} />Schedule Migration</label>
                        <Datetime id="PageMigrationScheduler_scheduleTime"
                            input={this.state.dateTimeInput}
                            onChange={this.handleDateTimeInput}
                            isValidDate={this.validDate }
                        />
                  </td>
                  <td width="40%"/>
                </tr>

              </tbody>
            </table>

        </div>

    );
  }
}
