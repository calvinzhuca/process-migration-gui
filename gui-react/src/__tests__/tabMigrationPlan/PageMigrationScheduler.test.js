import renderer from "react-test-renderer";
import React from "react";
import PageMigrationScheduler from "../../component/tabMigrationPlan/wizardExecuteMigration/PageMigrationScheduler";

test("PageMigrationScheduler renders correctly using snapshot", () => {
  const myMock = jest.fn();
  const tree = renderer
    .create(
      <PageMigrationScheduler
        setCallbackUrl={myMock}
        setScheduleStartTime={myMock}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
