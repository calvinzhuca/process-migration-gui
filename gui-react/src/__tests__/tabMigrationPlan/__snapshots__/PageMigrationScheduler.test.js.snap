// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageMigrationScheduler renders correctly using snapshot 1`] = `
<div
  className="form-horizontal"
>
  <table
    border="0"
    height="30%"
    width="100%"
  >
    <tbody>
      <tr>
        <td
          width="30%"
        />
        <td
          align="left"
          width="40%"
        >
          Callback URL: 
          <input
            name="callbackUrl"
            onChange={[Function]}
            type="text"
          />
        </td>
        <td
          width="40%"
        />
      </tr>
      <tr>
        <td
          width="30%"
        />
        <td
          align="left"
          width="40%"
        >
          <input
            name="timeType"
            onClick={[Function]}
            type="radio"
            value="1"
          />
          Run Migration Now
        </td>
        <td
          width="30%"
        />
      </tr>
      <tr>
        <td
          width="30%"
        />
        <td
          align="left"
          width="40%"
        >
          <input
            defaultChecked={true}
            name="timeType"
            onClick={[Function]}
            type="radio"
            value="2"
          />
          Schedule Migration
          <div
            className="rdt"
          >
            <input
              className="form-control"
              onChange={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              type="text"
              value=""
            />
            <div
              className="rdtPicker"
            >
              <div
                className="rdtDays"
              >
                <table>
                  <thead>
                    <tr>
                      <th
                        className="rdtPrev"
                        onClick={[Function]}
                      >
                        <span>
                          ‹
                        </span>
                      </th>
                      <th
                        className="rdtSwitch"
                        colSpan={5}
                        data-value={2}
                        onClick={[Function]}
                      >
                        March 2019
                      </th>
                      <th
                        className="rdtNext"
                        onClick={[Function]}
                      >
                        <span>
                          ›
                        </span>
                      </th>
                    </tr>
                    <tr>
                      <th
                        className="dow"
                      >
                        Su
                      </th>
                      <th
                        className="dow"
                      >
                        Mo
                      </th>
                      <th
                        className="dow"
                      >
                        Tu
                      </th>
                      <th
                        className="dow"
                      >
                        We
                      </th>
                      <th
                        className="dow"
                      >
                        Th
                      </th>
                      <th
                        className="dow"
                      >
                        Fr
                      </th>
                      <th
                        className="dow"
                      >
                        Sa
                      </th>
                    </tr>
                  </thead>
                  <tbody>
                    <tr>
                      <td
                        className="rdtDay rdtOld rdtDisabled"
                        data-value={24}
                      >
                        24
                      </td>
                      <td
                        className="rdtDay rdtOld rdtDisabled"
                        data-value={25}
                      >
                        25
                      </td>
                      <td
                        className="rdtDay rdtOld rdtDisabled"
                        data-value={26}
                      >
                        26
                      </td>
                      <td
                        className="rdtDay rdtOld rdtDisabled"
                        data-value={27}
                      >
                        27
                      </td>
                      <td
                        className="rdtDay rdtOld rdtDisabled"
                        data-value={28}
                      >
                        28
                      </td>
                      <td
                        className="rdtDay rdtDisabled"
                        data-value={1}
                      >
                        1
                      </td>
                      <td
                        className="rdtDay rdtDisabled"
                        data-value={2}
                      >
                        2
                      </td>
                    </tr>
                    <tr>
                      <td
                        className="rdtDay rdtDisabled"
                        data-value={3}
                      >
                        3
                      </td>
                      <td
                        className="rdtDay rdtDisabled"
                        data-value={4}
                      >
                        4
                      </td>
                      <td
                        className="rdtDay rdtDisabled"
                        data-value={5}
                      >
                        5
                      </td>
                      <td
                        className="rdtDay rdtDisabled"
                        data-value={6}
                      >
                        6
                      </td>
                      <td
                        className="rdtDay rdtDisabled"
                        data-value={7}
                      >
                        7
                      </td>
                      <td
                        className="rdtDay rdtDisabled"
                        data-value={8}
                      >
                        8
                      </td>
                      <td
                        className="rdtDay rdtToday"
                        data-value={9}
                        onClick={[Function]}
                      >
                        9
                      </td>
                    </tr>
                    <tr>
                      <td
                        className="rdtDay"
                        data-value={10}
                        onClick={[Function]}
                      >
                        10
                      </td>
                      <td
                        className="rdtDay"
                        data-value={11}
                        onClick={[Function]}
                      >
                        11
                      </td>
                      <td
                        className="rdtDay"
                        data-value={12}
                        onClick={[Function]}
                      >
                        12
                      </td>
                      <td
                        className="rdtDay"
                        data-value={13}
                        onClick={[Function]}
                      >
                        13
                      </td>
                      <td
                        className="rdtDay"
                        data-value={14}
                        onClick={[Function]}
                      >
                        14
                      </td>
                      <td
                        className="rdtDay"
                        data-value={15}
                        onClick={[Function]}
                      >
                        15
                      </td>
                      <td
                        className="rdtDay"
                        data-value={16}
                        onClick={[Function]}
                      >
                        16
                      </td>
                    </tr>
                    <tr>
                      <td
                        className="rdtDay"
                        data-value={17}
                        onClick={[Function]}
                      >
                        17
                      </td>
                      <td
                        className="rdtDay"
                        data-value={18}
                        onClick={[Function]}
                      >
                        18
                      </td>
                      <td
                        className="rdtDay"
                        data-value={19}
                        onClick={[Function]}
                      >
                        19
                      </td>
                      <td
                        className="rdtDay"
                        data-value={20}
                        onClick={[Function]}
                      >
                        20
                      </td>
                      <td
                        className="rdtDay"
                        data-value={21}
                        onClick={[Function]}
                      >
                        21
                      </td>
                      <td
                        className="rdtDay"
                        data-value={22}
                        onClick={[Function]}
                      >
                        22
                      </td>
                      <td
                        className="rdtDay"
                        data-value={23}
                        onClick={[Function]}
                      >
                        23
                      </td>
                    </tr>
                    <tr>
                      <td
                        className="rdtDay"
                        data-value={24}
                        onClick={[Function]}
                      >
                        24
                      </td>
                      <td
                        className="rdtDay"
                        data-value={25}
                        onClick={[Function]}
                      >
                        25
                      </td>
                      <td
                        className="rdtDay"
                        data-value={26}
                        onClick={[Function]}
                      >
                        26
                      </td>
                      <td
                        className="rdtDay"
                        data-value={27}
                        onClick={[Function]}
                      >
                        27
                      </td>
                      <td
                        className="rdtDay"
                        data-value={28}
                        onClick={[Function]}
                      >
                        28
                      </td>
                      <td
                        className="rdtDay"
                        data-value={29}
                        onClick={[Function]}
                      >
                        29
                      </td>
                      <td
                        className="rdtDay"
                        data-value={30}
                        onClick={[Function]}
                      >
                        30
                      </td>
                    </tr>
                    <tr>
                      <td
                        className="rdtDay"
                        data-value={31}
                        onClick={[Function]}
                      >
                        31
                      </td>
                      <td
                        className="rdtDay rdtNew"
                        data-value={1}
                        onClick={[Function]}
                      >
                        1
                      </td>
                      <td
                        className="rdtDay rdtNew"
                        data-value={2}
                        onClick={[Function]}
                      >
                        2
                      </td>
                      <td
                        className="rdtDay rdtNew"
                        data-value={3}
                        onClick={[Function]}
                      >
                        3
                      </td>
                      <td
                        className="rdtDay rdtNew"
                        data-value={4}
                        onClick={[Function]}
                      >
                        4
                      </td>
                      <td
                        className="rdtDay rdtNew"
                        data-value={5}
                        onClick={[Function]}
                      >
                        5
                      </td>
                      <td
                        className="rdtDay rdtNew"
                        data-value={6}
                        onClick={[Function]}
                      >
                        6
                      </td>
                    </tr>
                  </tbody>
                  <tfoot>
                    <tr>
                      <td
                        className="rdtTimeToggle"
                        colSpan={7}
                        onClick={[Function]}
                      >
                        12:00 AM
                      </td>
                    </tr>
                  </tfoot>
                </table>
              </div>
            </div>
          </div>
        </td>
        <td
          width="40%"
        />
      </tr>
    </tbody>
  </table>
</div>
`;
